### PR TITLE
Add support for inline subscript

### PR DESCRIPTION
**Pull request by nlisgo**
Improves: https://github.com/libero/libero/issues/106

### CODE REVIEW

File: extensions/inline/sub.rng [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <empty/>
+    </define>
+
+    <define name="libero.sub.limited.content">
+        <ref name="libero.inline.limited"/>
+    </define>
+
+    <sch:pattern>
+        <sch:rule context="libero:sub">
+            <sch:assert test="not(ancestor::libero:sub)">
```
Comment (thewilkybarkid) [inline extensions/inline/sub.rng]: Probably want to not allow it inside `sup` too (and the reverse). We can merge this one first though and add it in the other.

**Comment by giorgiosironi** [comment]
See also https://github.com/libero/schemas/pull/18#issuecomment-464003022

File: extensions/inline/sub.rng (hunk 1 above)
Comment (nlisgo) [reply extensions/inline/sub.rng]: @thewilkybarkid is there a reason why we would nest any elements within `sub` or `sup`?

Comment (thewilkybarkid) [reply extensions/inline/sub.rng]: I meant you probably shouldn't have subscript inside superscript...

Comment (thewilkybarkid) [reply extensions/inline/sub.rng]: Misunderstood this, question is like https://github.com/libero/schemas/pull/18#issuecomment-464003022.

Not aware of any scenarios where you would, so safe to not allow it. The extensibility model allows you to add it in if needed (so `libero.sub.full.content` would need to contain `<empty/>`).

File: extensions/inline/sub.rng [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<grammar ns="http://libero.pub" xmlns="http://relaxng.org/ns/structure/1.0"
+    datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron">
+
+    <sch:ns prefix="libero" uri="http://libero.pub"/>
+
+    <define name="libero.inline.full.highlighting" combine="choice">
```
Comment (thewilkybarkid) [inline extensions/inline/sub.rng]: These aren't highlightling, suggest we either follow JATS and use `subsup`, or maybe `positioning` is better? (Maybe sounds too broad though.)

File: tests/extensions/inline/sub/full/invalid/extra-content.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="../extra-content.rng" schematypens="http://relaxng.org/ns/structure/1.0"?>
+<?xml-model href="../extra-content.rng" schematypens="http://purl.oclc.org/dsdl/schematron"?>
```
Comment (thewilkybarkid) [inline tests/extensions/inline/sub/full/invalid/extra-content.xml]: This case/schema doesn't do anything anymore as it no longer uses `libero.inline.full` (`unknown-element` already covers there being an unknown element).

File: tests/extensions/inline/sub/full/invalid/nested.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="../basic.rng" schematypens="http://relaxng.org/ns/structure/1.0"?>
+<?xml-model href="../basic.rng" schematypens="http://purl.oclc.org/dsdl/schematron"?>
+<?expected-error line="7" message="Did not expect element sub there"?>
```
Comment (thewilkybarkid) [inline tests/extensions/inline/sub/full/invalid/nested.xml]: Same as https://github.com/libero/schemas/pull/17#discussion_r257590946

File: tests/extensions/inline/sub/full/invalid/extra-content.xml (hunk 3 above)
Comment (nlisgo) [reply tests/extensions/inline/sub/full/invalid/extra-content.xml]: As I understand it we can get rid of the following tests:

- tests/extensions/inline/sup/full/invalid/extra-content.xml
- tests/extensions/inline/sup/full/invalid/nested.xml
- tests/extensions/inline/sup/full/invalid/nested-deep.xml
- tests/extensions/inline/sup/limited/invalid/extra-content.xml
- tests/extensions/inline/sup/limited/invalid/nested.xml
- tests/extensions/inline/sup/limited/invalid/nested-deep.xml

All the above cases/schemas are covered by:

- tests/extensions/inline/sup/full/invalid/unknown-element.xml
- tests/extensions/inline/sup/limited/invalid/unknown-element.xml

**Review by thewilkybarkid — APPROVED** [review]

**Pull request merged**